lib/bolt/cli.rb

```diff
@@ -676,18 +689,27 @@ def initialize_project
 
       # Warn the user if the project directory already exists. We don't error here since users
       # might not have installed any modules yet.
+      # If both bolt.yaml and bolt-project.yaml exist, this will just warn
+      # about bolt-project.yaml and subsequent Bolt actions will warn about
+      # both files existing
       if config.exist?
-        @logger.warn "Found existing project directory at #{project}"
+        @logger.warn "Found existing project directory at #{project}. Skipping file creation."
+      # This won't get called if bolt-project.yaml exists
+      elsif old_config.exist?
+        @logger.warn "Found existing #{old_config.basename} at #{project}. "\
+        "Attempting to update to #{config.basename}."
+        update_project_config(old_config)
```

Comment:
This should error with a helpful message instead of attempting to convert the `bolt.yaml` file to `bolt-project.yaml`.

Reply:
I was using the principle of "fix it instead of telling the user it's wrong and that they should fix it" - is there a reason you think it should error instead?

Reply:
My thinking is that what's "wrong" here isn't which file exists on disk, but the command the user typed. It's not that they have a bolt.yaml when they need a bolt-project.yaml, but that they tried to use `bolt project init` in a directory that is already a project. I'm on the fence about it though.

Reply:
It's mostly because in planning we had decided that it should error if there's a `bolt.yaml`. I'm not strictly opposed to having it convert a `bolt.yaml` if it's able to. However, if it does convert it I think we might as well add the `name` key for the user. It seems a little strange to go through the effort of converting the file for them but then have them manually enter the name.

Reply:
Hm, I initially didn't add 'name' because I was thinking the way to do it would be to read in yaml, add the key, then write it out, which removes comments and reorders things. But I think we might be able to safely just append the name to the config file...that would override a name if it was already there, but that'd be a particularly weird case.